benchmarks/base_classes.py

```diff
@@ -344,3 +347,72 @@ class T2IAdapterSDXLBenchmark(T2IAdapterBenchmark):
 
     def __init__(self, args):
         super().__init__(args)
+
+
+class BaseBenchmarkTestCase:
+    model_class = None
+    pretrained_model_name_or_path = None
+    model_class_name = None
+
+    def __init__(self):
+        super().__init__()
+
+    def get_result_filepath(self, suffix):
+        name = (
+            self.model_class_name
+            + "_"
+            + self.pretrained_model_name_or_path.replace("/", "_")
+            + "_"
+            + f"{suffix}.csv"
+        )
+        filepath = os.path.join(BASE_PATH, name)
+        return filepath
+
+
+class AutoencoderKLBenchmark(BaseBenchmarkTestCase):
```

Comment:
Should we let the users define `dummy_inputs()` per model class here? And then we could let them implement their own function that needs to be benchmarked. 

So, `BaseBenchmarkTestCase` could then have a method `benchmark()`:

```py
def benchmark(...):
	time = benchmark_fn(self.run_decode, self.model, tensor)
    memory = bytes_to_giga_bytes(torch.cuda.max_memory_allocated()) # should this be allocated?
    benchmark_info = BenchmarkInfo(time=time, memory=memory)
    
    csv_dict = generate_csv_dict_model(
        model_cls=self.model_class_name, ckpt=self.pretrained_model_name_or_path, benchmark_info=benchmark_info, **kwargs,
    )
    print(f"{self.model_class_name} decode - shape: {list(tensor.shape)}, time: {time}, memory: {memory}")
    return csv_dict
```